contracts/src/v0.8/dev/automation/2_1/KeeperRegistryLogicA2_1.sol

```diff
@@ -417,6 +414,9 @@ contract KeeperRegistryLogicA2_1 is
       bytes[] memory offchainConfigs
     ) = abi.decode(encodedUpkeeps, (uint256[], Upkeep[], address[], bytes[], bytes[], bytes[]));
     for (uint256 idx = 0; idx < ids.length; idx++) {
+      if (address(upkeeps[idx].forwarder) == ZERO_ADDRESS) {
+        upkeeps[idx].forwarder = new AutomationForwarder(ids[idx], upkeeps[idx].target, address(this));
+      }
```

Comment:
Was the forwarder not assigned in v2_0 then, or was it assigned elsewhere?

Reply:
We didn't have one - it's a new feature in v2.1